ingress/controllers/nginx/README.md

```diff
@@ -149,6 +150,23 @@ By default the controller redirects (301) to HTTPS if there is a TLS Ingress rul
 To disable this behavior use `hsts=false` in the NGINX config map.
 
 
+### Automated Certificate Management with Kube-Lego
+
+[Kube-Lego] automatically requests missing certificates or expired from
+[Let's Encrypt] by monitoring ingress resources and its referenced secrets. To
+enable this for an ingress resource you have to add an annotation:
+
+```
+kubectl annotate ing ingress-demo kubernetes.io/tls-acme="true"
```

Comment:
Yeah keeping it in a secret is exactly the sort of api i'd expect. My questions was: say I write a haproxy backend for ingress today, does it need to do anything special to augment kube-lego (like serve .well-known/acme-challenge/), or is kube-lego capable of satisfying the acme challenges on its own?